lib/src/deploy.rs

```diff
@@ -552,6 +573,12 @@ fn origin_from_imageref(imgref: &ImageReference) -> Result<glib::KeyFile> {
     Ok(origin)
 }
 
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub(crate) struct StageOptions {
```

Comment:
Not opposed but this feels slightly overkill at this point - we don't have *that* many parameters to the function.

(And if we did we may as well move the progress writer in?)

I think a simple `kexec: bool` would be OK no?